src/views/partials/measure-preview-table.ejs

```diff
@@ -0,0 +1,43 @@
+<table class="govuk-table">
+    <thead class="govuk-table__head">
+    <tr class="govuk-table__row">
+        <% headers.forEach(function(cell, idx) { %>
+            <th scope="col" class="govuk-table__header">
+                <% if (t(`publish.measure_review.column_headers.${cell.name.toLowerCase()}`) === `publish.measure_review.column_headers.${cell.name.toLowerCase()}`) { %>
+                    <%= cell.name %>
+                <% } else { %>
+                    <%= t(`publish.measure_review.column_headers.${cell.name.toLowerCase()}`) %>
+                <% } %>
+            </th>
+        <% }); %>
+    </tr>
+    </thead>
+    <tbody>
+    <% data.forEach(function(row) { %>
+        <% if (locals.langCol > -1 && row[locals.langCol] !== i18n.language.toLowerCase() ) return; %>
+        <tr>
+            <% row.forEach(function(cell, index) { %>
+                <td class="govuk-table__cell">
+                    <% switch (locals.headers[index].name.toLowerCase()) {
+                        case 'start_date':
+                    case 'end_date': %>
+                    <%= locals.dateFormat(locals.parseISO(cell.split('T')[0]), 'do MMMM yyyy') %>
+                    <% break;
+                    case 'date_type': %>
+                    <%= t(`publish.measure_review.year_type.${cell}`) %>
+                    <% break;
+                    default: %>
+                    <% if (cell) { %>
+                        <% if (t(`publish.measure_review.column_values.${cell.toString().toLowerCase()}`) === `publish.measure_review.column_values.${cell.toString().toLowerCase()}`) { %>
+                            <%= cell %>
+                        <% } else { %>
+                            <%= t(`publish.measure_review.column_values.${cell.toString().toLowerCase()}`) %>
+                        <% } %>
+                    <% } %>
+                    <% } %>
+                </td>
+            <% }); %>
```

Comment:
The nesting is a bit all over the place here and hard to follow - tbh if we need to do this much conditional formatting to the data we should probably do it in the controller before passing it to the view, and leave the view as purely playback.

I realise this might've just moved from elsewhere but a lot of stuff got hand-waved through because it was urgent for MVP.

Reply:
So the tables are mostly formatting the tables which come from the backend... You could just render the "raw" view of the table but then you loose some of the friendliness of the table.  I don't see there being any gain moving it to the controller level as in effect you'd just end up rendering the whole table in the controller and passing it to the view.  Rather than just building the view based on the raw table.